sentry-android-core/src/main/java/io/sentry/android/core/performance/TimeSpan.java

```diff
@@ -90,7 +91,8 @@ public long getStartTimestampMs() {
    */
   public @Nullable SentryDate getStartTimestamp() {
     if (hasStarted()) {
-      return new SentryLongDate(DateUtils.millisToNanos(getStartTimestampMs()));
+      return new SentryNanotimeDate(
+          DateUtils.nanosToDate(DateUtils.millisToNanos(getStartTimestampMs())), System.nanoTime());
```

Comment:
`m`: since this is just a getter, it will return a different `SentryDate` every time it gets invoked. Would it make more sense to just provide `0` for the nanos?
```suggestion
          DateUtils.nanosToDate(DateUtils.millisToNanos(getStartTimestampMs())), 0);
```
If we need nano precision, we could store the `System.nanoTime()` within the `start() { }` block and then re-use it here later.

Reply:
we need nano precision, as in the `FrameMetricsCollector` we call `date.diff`, which uses that value and compares it to the value of other dates, which are all `System.nanotime()`.
I'm setting it during `start()`